plugins/database/hana/hana_test.go

```diff
@@ -22,122 +22,231 @@ func TestHANA_Initialize(t *testing.T) {
 		"connection_url": connURL,
 	}
 
-	db := new()
-	_, err := db.Init(context.Background(), connectionDetails, true)
-	if err != nil {
-		t.Fatalf("err: %s", err)
+	initReq := dbplugin.InitializeRequest{
+		Config:           connectionDetails,
+		VerifyConnection: true,
 	}
 
-	if !db.Initialized {
-		t.Fatal("Database should be initialized")
-	}
-
-	err = db.Close()
-	if err != nil {
-		t.Fatalf("err: %s", err)
+	db := new()
+	initResp := dbtesting.AssertInitialize(t, db, initReq)
+	if initResp.Config == nil {
```

Comment:
FWIW, I've seen this style of checking the initialize response in a few places: https://github.com/hashicorp/vault/blob/a69ee0f65a28527376ab3d1cac89914a96406e30/plugins/database/mongodb/mongodb_test.go#L35-L48

Reply:
Updated